### DISCUSSION TITLE
Several sizes for the Balance component

### DISCUSSION THREAD

**Discussion by rizoom** [scaffold-eth/scaffold-eth-2#373] [Ideas]
While working on migrating some view from SE1 to SE2, I notice that it was easy to customize the size of the balance component in SE1, and it appears to not be the case anymore in SE2.

**Ex: Multisig wallet page in SE1 with a big balance component**
<img width="329" alt="image" src="https://github.com/scaffold-eth/scaffold-eth-2/assets/1694104/46818ef6-c440-44bd-b628-413c2ece7865">

I am starting this discussion to assess what could be added to the Balance component, and then I'd be happy to propose a PR for it.

👉 My proposal would be to add a size props (same as in address component) to allow for these [font sizes](https://tailwindcss.com/docs/font-size)
💡 Other than size, any idea of other useful props to add for this component?

**Comment by rizoom** [comment]
Corresponding example in SE2, highlighting the need for a bigger balance:
<img width="413" alt="image" src="https://github.com/scaffold-eth/scaffold-eth-2/assets/1694104/946ab213-7bc5-40f1-901c-b5416d623bd1">

**Comment by sverps** [comment]
Hey @rizoom, thanks for the suggestion. You can already modify the fontsize by either placing the `Balance` component inside a parent that has a certain fontsize (e.g. `h1` element or any styled parent), or you can set the fontsize on the `className` prop of the `Balance` component itself.

In checking this, I did find a small issue that the ETH / $ part remained small, so I created a PR to resolve this as well (#375).

**Reply by technophile-04** [reply]
Samuel, what are your views on adding the `size` prop? 

Like the className="text-2xl" etc does the job nicely but just thinking if it's intuitive for the users using the component because people who aren't familiar with a tailwind won't click up to them

**Reply by sverps** [reply]
It's not tailwind specific, it's how css works, text is displayed at the current font-size, which looks to the inner most parent that defines a font-size. So there's a bunch of different ways the font size could be derived, similarly to how text color and a bunch of other css inheritances work.

So with that in mind, I'd prefer to not add too many customizer props on components in order to keep code complexity low and keep consistency across components. So I think relying on a className (with plenty of examples from the tailwind classes that are used) and default css behavior is good enough.

The alternative (to keep consistent behavior) is to go through all the components and create a set of modifiers that should work on all of them.

**Reply by technophile-04** [reply]
Yup 

Actually, I was not talking about `em` 😅 but the DevX while using the component : 

```js

<Blance className="text-2xl" /> 

// ---VS---

<Balnace size="2xl" />
```
like passing "text-2xl" won't click to the users (who aren't familiar with a tailwind like which tw class to use) unless they go-to the definition of the component and see where the `className` prop is used. 

This just came to my mind because we recently added a `size` prop to the `Address` component at #362 , but yeah I agree with your 2nd point

**Reply by sverps** [reply]
I think the difference with the `Address` component is that there it's not just text, there's the blockie or ENS avatar and the copy icon, so the whole thing needs to scale and it doesn't only look at the font size I guess.

**Reply by carletex** [reply]
Love this discussion, thank you! :)

> I'd prefer to not add too many customizer props on components in order to keep code complexity low and keep consistency across components.

Agree. But adding the `size` prop across all the components (balance, inputs) won't bother me either.

Consistency would be the key for me. Rn we have:

- Address. Size prop
- Balance: classname
- Inputs: nothing :D

So not very consistent.

I'd keep the discussion open, see what we like most and unify. 

I personally like the approach we have in the Address component. We could extract this (size prop + blockie mapping + avatar mapping?) somewhere and use it from different components.

![image](https://github.com/scaffold-eth/scaffold-eth-2/assets/2486142/15b3e585-ea88-4ad8-9d2d-bfefc92cde6d)

e.g. address input also has blockie / avatars.


P.S. Not feeling super strong about this approach!!

**Reply by sverps** [reply]
I'm having an internal conflict between leveraging tailwind's classes (keeping code simpler and more flexible) vs adding some modifier props (making our components maybe a bit more noob-friendly and intuitive).

I guess it's not urgent and we can wait a bit more and see if any additional feedback arises. Probably not a huge issue anyway.

**Comment by rizoom** [comment]
Agree that consistency is key.
So as to whether adding specific props to components (with consistency in mind), I'd say it might depend on what the vision for SE2 is.
Is it to provide and maintain a set of structured, documented, powerful and easy to use components through SE2, or on the contrary keep it simple and let builders use the front-end app as a starting point from which they will customize existing components for they needs and experiments?
If you were to go with the "rich and structured" component approach, you could even consider moving some core components in an external package, such as the cool address and input components. I guess it would help SE2 branches/forks to easily update to the latest version of the core components.

As you said, not the most important or urgent topic, it's more about consistency and long term vision which is an open topic.
So, regarding the size of the Balance, thanks for the PR! I'll mark this discussion as solved.
